marvin-ai/marvin-engine-executor
# Allowing json message on /predictor + adding support to prediction message schema validation

Fixes #61 

The predictor request body can now receive a json in the field message and params. An string is not accepted anymore in both.

Example of a valid input for /predictor:
```
{
	"message": {
		"firstName": "Lucas",
		"lastName": "bm",
		"test": "te"
	}
}
```

Also, now it is possible to validate the message value against a pre-defined schema. (ref to http://json-schema.org/draft-04/json-schema-validation.html)

A valid schema should exists with the name MessageSchema.json in the engine root. A valid schema definition should look like:
```
{
    "title": "PredictMessage",
    "type": "object",
    "properties": {
        "firstName": {
            "type": "string"
        },
        "lastName": {
            "type": "string"
        },
        "age": {
            "description": "Age in years",
            "type": "integer",
            "minimum": 0
        }
    },
    "required": ["firstName", "lastName"]
}
```

### How to test:
- Create the file <actionName>-message.schema in the root of your engine (Eg. predictor-message.schema).
- Paste the schema json above in the file.
- Try to perform a request to /predictor with the body:
```
{
	"message": {
		"firstName": "Lucas",
		"test": "te"
	}
}
```
- Check that a status code 400 is presented and a message stating that the lastName field is missing is shown.
